src/os/bluestore/bluestore_types.h

```diff
@@ -196,6 +196,7 @@ struct bluestore_blob_t {
   }
 
   vector<bluestore_pextent_t> extents;///< raw data position on device
+  uint32_t compressed_length_orig = 0;///< original length of compressed blob if any
```

Comment:
I'd prefer to highlight that this member is applied to compressed blobs only. And we ensure its' persistence in this case only. 'length' hides such semantic IMHO.